.gitmodules

```diff
@@ -31,9 +31,9 @@
 [submodule "src/stdarch"]
 	path = src/stdarch
 	url = https://github.com/rust-lang/stdarch.git
-[submodule "src/doc/rustc-guide"]
-	path = src/doc/rustc-guide
-	url = https://github.com/rust-lang/rustc-guide.git
+[submodule "src/doc/rustc-dev-guide"]
+	path = src/doc/rustc-dev-guide
+	url = https://github.com/rust-lang/rustc-dev-guide.git
```

Comment:
Is this change handled gracefully by our build system or do I need to change something else? cc @Mark-Simulacrum

Reply:
I think it's fine. But perhaps also git mv to rename the directory.

Reply:
We should update the submodule after the rename. As soon as we do I can fix this PR.

Reply:
I believe this should be fine, though I'm personally uncertain whether something else needs to change (i.e., our tooling should be fine but I don't know what git'll want).